src/canvasObject.js

```diff
@@ -1,159 +1,128 @@
 'use strict';
 
 require('openseadragon');
+var ImageResource = require('./ImageResource');
 
 var CanvasObject = function(config, dispatcher) {
-  this.fullyOpened = config.fullyOpened || false;
-  this.visible = config.visible || true; // todo: this is not used yet. Do we need it?
+  var self = this;
```

Comment:
I just noticed that the dispatcher is a second argument...it should just be one of the properties of the config.